src/pages/AccountDetails/AccountTokensTable/helpers/sortTokens.ts

```diff
@@ -0,0 +1,128 @@
+import BigNumber from 'bignumber.js';
+
+import { LOW_LIQUIDITY_DISPLAY_TRESHOLD } from 'appConstants';
+import { formatAmount } from 'helpers';
+import { TokenType, SortOrderEnum } from 'types';
+
+export enum SortTokenFieldEnum {
+  name = 'name',
+  balance = 'balance',
+  price = 'price',
+  value = 'value',
+  portofolioPercent = 'portofolioPercent'
+}
+
+export interface ProcessedTokenType extends TokenType {
+  portofolioPercentage: BigNumber;
+}
+
+export interface SortTokensType {
+  field?: SortTokenFieldEnum;
+  order?: SortOrderEnum;
+  tokens: ProcessedTokenType[];
+  tokenBalance?: string;
+}
+
+const getTokenDisplayValue = ({
+  valueUsd,
+  isLowLiquidity
+}: ProcessedTokenType) => {
+  if (
+    valueUsd &&
+    (!isLowLiquidity ||
+      new BigNumber(valueUsd).isLessThan(LOW_LIQUIDITY_DISPLAY_TRESHOLD))
+  ) {
+    return new BigNumber(valueUsd);
+  }
+
+  return new BigNumber(0);
+};
+
+const getPortofolioPercent = ({
+  token,
+  tokenBalance
+}: {
+  token: ProcessedTokenType;
+  tokenBalance: string;
+}) => {
+  return token.valueUsd && tokenBalance
+    ? new BigNumber(token.valueUsd).dividedBy(tokenBalance).times(100)
+    : new BigNumber(0);
+};
+
+export const sortTokens = ({
```

Comment:
I'd try to use more memoization in these user actions

Reply:
Both `filterTokens` and `sortTokens` are only used in a hook that memoizes the response. They were split into separate files for better readability on that respective hook (`useProcessTokens`)